src/main/java/net/mcreator/ui/modgui/VillagerProfessionGUI.java

```diff
@@ -0,0 +1,207 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.modgui;
+
+import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.element.types.VillagerProfession;
+import net.mcreator.minecraft.ElementUtil;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.SearchableComboBox;
+import net.mcreator.ui.component.util.ComboBoxUtil;
+import net.mcreator.ui.component.util.ComponentUtils;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.dialogs.TextureImportDialogs;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.laf.renderer.WTextureComboBoxRenderer;
+import net.mcreator.ui.minecraft.MCItemHolder;
+import net.mcreator.ui.minecraft.SoundSelector;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.ui.validation.ValidationGroup;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VComboBox;
+import net.mcreator.ui.validation.component.VTextField;
+import net.mcreator.ui.validation.validators.TextFieldValidator;
+import net.mcreator.ui.validation.validators.UniqueNameValidator;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.ListUtils;
+import net.mcreator.util.StringUtils;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.File;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+public class VillagerProfessionGUI extends ModElementGUI<VillagerProfession> {
+
+	private final VTextField displayName = new VTextField(30);
+	private final MCItemHolder pointOfInterest = new MCItemHolder(mcreator, ElementUtil::loadBlocks);
+	private final SoundSelector actionSound = new SoundSelector(mcreator);
+	private final JComboBox<String> hat = new JComboBox<>(new String[] { "None", "Partial", "Full" });
+	private final VComboBox<String> professionTextureFile = new SearchableComboBox<>();
+	private final VComboBox<String> zombifiedProfessionTextureFile = new SearchableComboBox<>();
+
+	private final ValidationGroup page1group = new ValidationGroup();
+
+	public VillagerProfessionGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
+		super(mcreator, modElement, editingMode);
+		this.initGUI();
+		super.finalizeGUI();
+	}
+
+	@Override protected void initGUI() {
+		professionTextureFile.setRenderer(
+				new WTextureComboBoxRenderer.TypeTextures(mcreator.getWorkspace(), TextureType.ENTITY));
+		professionTextureFile.setPrototypeDisplayValue("XXXXXXXXXXXXXXXXXXXXXXXXXX");
+		zombifiedProfessionTextureFile.setRenderer(
+				new WTextureComboBoxRenderer.TypeTextures(mcreator.getWorkspace(), TextureType.ENTITY));
+		zombifiedProfessionTextureFile.setPrototypeDisplayValue("XXXXXXXXXXXXXXXXXXXXXXXXXX");
+
+		ComponentUtils.deriveFont(displayName, 16);
+		ComponentUtils.deriveFont(hat, 16);
```

Comment:
Question to maintainers: are these two needed?

Reply:
Yes, we use this font size for those cases in other places too